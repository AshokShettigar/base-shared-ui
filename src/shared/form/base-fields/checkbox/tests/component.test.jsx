import { shallow } from 'enzyme';

import Checkbox from '../component';

describe('Checkbox component', () => {
    describe('should render component with', () => {
        it('default props', () => {
            const wrapper = shallow(<Checkbox />);

            expect(wrapper).toMatchSnapshot();
        });

        it('id prop', () => {
            const wrapper = shallow(<Checkbox id="id" />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('name prop', () => {
            const wrapper = shallow(<Checkbox name="name" />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('value prop', () => {
            const wrapper = shallow(<Checkbox value="value" />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('children prop', () => {
            const wrapper = shallow(<Checkbox>Children</Checkbox>).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('error prop', () => {
            const wrapper = shallow(<Checkbox error="error" />).dive();

            expect(wrapper).toMatchSnapshot();
        });


        it('autoFocus prop', () => {
            const wrapper = shallow(<Checkbox autoFocus />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('readOnly prop', () => {
            const wrapper = shallow(<Checkbox readOnly />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('checked prop', () => {
            const wrapper = shallow(<Checkbox checked />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('disabled and checked props', () => {
            const wrapper = shallow(<Checkbox checked disabled />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('disabled prop', () => {
            const wrapper = shallow(<Checkbox disabled />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('multiple prop', () => {
            const wrapper = shallow(<Checkbox multiple />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('multiple and disabled props', () => {
            const wrapper = shallow(<Checkbox disabled multiple />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('multiple and checked props', () => {
            const wrapper = shallow(<Checkbox checked multiple />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('multiple and indeterminate props', () => {
            const wrapper = shallow(<Checkbox indeterminate multiple />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('indeterminate and disabled prop', () => {
            const wrapper = shallow(<Checkbox indeterminate disabled />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('indeterminate prop', () => {
            const wrapper = shallow(<Checkbox indeterminate />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('class names', () => {
            const wrapper = shallow(<Checkbox className="className" labelClassName="labelClassName" />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('refField prop', () => {
            const wrapper = shallow(<Checkbox refField={() => {}} />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('onChange prop', () => {
            const wrapper = shallow(<Checkbox onChange={() => {}} />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('hidden component', () => {
            const wrapper = shallow(<Checkbox hidden />).shallow();

            expect(wrapper).toMatchSnapshot();
        });
    });

    it('shouldn\'t render component', () => {
        const wrapper = shallow(<Checkbox rendered={false} />);

        expect(wrapper).toMatchSnapshot();
    });

    describe('setRef', () => {
        it('should called innerRef', () => {
            const innerRef = jest.fn();
            const wrapper = shallow(<Checkbox innerRef={innerRef} />).shallow();

            wrapper.instance().setRef('DOMNode');

            expect(wrapper.instance().checkboxRef).toEqual('DOMNode');
            expect(innerRef).toHaveBeenCalled();
        });

        it('shouldn\'t called innerRef', () => {
            const innerRef = jest.fn();
            const wrapper = shallow(<Checkbox />).shallow();

            wrapper.instance().setRef();

            expect(wrapper.instance().checkboxRef).toBeUndefined();
            expect(innerRef).not.toHaveBeenCalled();
        });
    });

    describe('onChange', () => {
        it('should called', () => {
            const onChange = jest.fn();
            const wrapper = shallow(<Checkbox name="name" value="value" onChange={onChange} />).shallow();

            wrapper.instance().checkboxRef = 'DOMNode';
            wrapper.instance().onChange();

            expect(onChange).toHaveBeenCalled();
        });

        it('should\'t called', () => {
            const onChange = jest.fn();
            const wrapper = shallow(<Checkbox />).shallow();

            wrapper.instance().checkboxRef = 'DOMNode';
            wrapper.instance().onChange();

            expect(onChange).not.toHaveBeenCalled();
        });
    });
});
