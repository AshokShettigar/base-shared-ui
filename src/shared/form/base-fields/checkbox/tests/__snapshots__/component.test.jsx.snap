// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render component with autoFocus prop 1`] = `
<span
  className="root"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={true}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with checked prop 1`] = `
<span
  className="root checked"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={true}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      >
        <span
          className="checkmarkGlyph"
        >
          <WithRender
            className="icon"
            glyph="checkmark"
            hidden={false}
            rendered={true}
            size="mini"
          />
        </span>
      </span>
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with children prop 1`] = `
<span
  className="root"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
      <span
        aria-checked={false}
        className="labelText"
        role="radio"
        tabIndex={-1}
      >
        Children
      </span>
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with class names 1`] = `
<span
  className="root className"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with default props 1`] = `
<Checkbox
  autoFocus={false}
  checked={false}
  disabled={false}
  hidden={false}
  indeterminate={false}
  multiple={false}
  readOnly={false}
/>
`;

exports[`Checkbox component should render component with disabled and checked props 1`] = `
<span
  className="root disabled checked"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={true}
        disabled={true}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      >
        <span
          className="checkmarkGlyph"
        >
          <WithRender
            className="icon"
            glyph="checkmark"
            hidden={false}
            rendered={true}
            size="mini"
          />
        </span>
      </span>
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with disabled prop 1`] = `
<span
  className="root disabled"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={true}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with error prop 1`] = `
<span
  className="root"
  data-component="Checkbox"
>
  <WithRender
    error="error"
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with hidden component 1`] = `
<div
  hidden={true}
>
  <Checkbox
    autoFocus={false}
    checked={false}
    disabled={false}
    hidden={true}
    indeterminate={false}
    multiple={false}
    readOnly={false}
  />
</div>
`;

exports[`Checkbox component should render component with id prop 1`] = `
<span
  className="root"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
      htmlFor="id"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with indeterminate and disabled prop 1`] = `
<span
  className="root disabled indeterminate"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={true}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      >
        <i
          className="indeterminateGlyph"
        />
      </span>
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with indeterminate prop 1`] = `
<span
  className="root indeterminate"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      >
        <i
          className="indeterminateGlyph"
        />
      </span>
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with multiple and checked props 1`] = `
<span
  className="root multiple checked"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={true}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      >
        <span
          className="checkmarkGlyph"
        >
          <WithRender
            className="icon"
            glyph="checkmark"
            hidden={false}
            rendered={true}
            size="mini"
          />
        </span>
      </span>
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with multiple and disabled props 1`] = `
<span
  className="root multiple disabled"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={true}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with multiple and indeterminate props 1`] = `
<span
  className="root multiple indeterminate"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      >
        <i
          className="indeterminateGlyph"
        />
      </span>
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with multiple prop 1`] = `
<span
  className="root multiple"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with name prop 1`] = `
<span
  className="root"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        name="name"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with onChange prop 1`] = `
<span
  className="root"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with readOnly prop 1`] = `
<span
  className="root readOnly"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with refField prop 1`] = `
<span
  className="root"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component should render component with value prop 1`] = `
<span
  className="root"
  data-component="Checkbox"
>
  <WithRender
    hidden={false}
    rendered={true}
  >
    <label
      className="label"
    >
      <input
        autoFocus={false}
        checked={false}
        disabled={false}
        hidden={true}
        onChange={[Function]}
        type="checkbox"
        value="value"
      />
      <span
        className="checkbox"
      />
    </label>
  </WithRender>
</span>
`;

exports[`Checkbox component shouldn't render component 1`] = `null`;
